HW_3/task_18.py

```diff
@@ -0,0 +1,28 @@
+# Задача 18: Требуется найти в массиве A[1..N] самый близкий по величине элемент к заданному числу X. 
+# Пользователь в первой строке вводит натуральное число N – количество элементов в массиве. 
+# В последующих  строках записаны N целых чисел Ai. Последняя строка содержит число X
+# *Пример:*
+# 5
+#     1 2 3 4 5
+#     6
+#     -> 5
+
+import random
+
+n = int(input("Введите длину массива: "))
+
+a = [random.randint(1, 10) for _ in range(n)]
+
+x = int(input("Введите число X: "))
+
+closest_num = a[0]
+min_diff = abs(x - a[0])
+
+for i in range(1, n):
+    diff = abs(x - a[i])
+    if diff < min_diff:
+        closest_num = a[i]
+        min_diff = diff

```

Comment:
Не обязательно использовать функцию `range`, можно напрямую проходиться по элементам списка.
```suggestion
for i in a[1:]:
    diff = abs(x - i)
    if diff < min_diff:
        closest_num = i
        min_diff = diff
```